### PR TITLE
fix(snippet): stop jump when current index is final one

### PR DESCRIPTION
**Pull request by glepnir**
Problem: The current snippet jump is circular but as lsp spec mentioned
                 the `$0` is final snippet section should stop when at it

Solution: when current tab index is 0 stopped.

Ref https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionClientCapabilities

### CODE REVIEW

File: runtime/lua/vim/snippet.lua [hunk 1]
```diff
@@ -674,7 +674,9 @@ end
 --- @return boolean
 function M.active(filter)
   local active = M._session ~= nil and M._session.bufnr == vim.api.nvim_get_current_buf()
-
+  if M._session.current_tabstop and M._session.current_tabstop.index == 0 then
```
Comment (justinmk) [inline runtime/lua/vim/snippet.lua:677]: when we have a reference, it's always worth a brief comment to provide that. it's a cheap way to help people reading this code in the future.

```suggestion
  -- "$0 defines the final tab stop".
  -- https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionClientCapabilities
  if M._session.current_tabstop and M._session.current_tabstop.index == 0 then
```

File: runtime/lua/vim/snippet.lua [hunk 2]
```diff
@@ -674,7 +674,9 @@ end
...
+    return active and false
```
Comment (justinmk) [inline runtime/lua/vim/snippet.lua:678]: `active and false` will always return falsy. 
```suggestion
    return false
```

File: runtime/lua/vim/snippet.lua (hunk 1 above)
Comment (glepnir) [reply runtime/lua/vim/snippet.lua:678]: because its too long.. exceeds the 80 column width 🤔

**Comment by glepnir** [comment]
It is not. Just try a snippet. It is circular. the test which i added in this Pr will be failed in master.
what

**Comment by clason** [comment]
I cannot replicate that. Are you sure you're using the built-in LSP snippet expansion? Please test with `nvim --clean`.

**Comment by glepnir** [comment]
I am sure it is but it happen when i use this code snippet , <C-X><C-O> works fine with it.

```lua
local function auto_trigger(bufnr, client)
  au(InsertCharPre, {
    buffer = bufnr,
    callback = function()
      if pumvisible() then
        return
      end
      local triggerchars = vim.tbl_get(
        client,
        'server_capabilities',
        'completionProvider',
        'triggerCharacters'
      ) or {}
      if vim.v.char:match('[%w_]') and not vim.list_contains(triggerchars, vim.v.char) then
        completion.trigger()
      end
    end,
  })
end
au('LspAttach', {
  callback = function(args)
    local bufnr = args.buf
    local client_id = args.data.client_id
    completion.enable(true, client_id, bufnr, {
      autotrigger = true,
      convert = function(item)
        return { abbr = item.label:gsub('%b()', ''), kind = '' }
      end,
    })
    local client = lsp.get_client_by_id(client_id)
    if not client then
      return
    end
    auto_trigger(bufnr, client)
  end,
})

```

**Pull request closed without merging**